design/ceph/multus-network.md

```diff
@@ -78,21 +78,71 @@ Nothing to do in particular since they don't use any service IPs.
-
-The holder daemonset is deleted when the CephCluster is marked for deletion. The termination of the holder pods trigger the teardown jobs to run, removing the multus interfaces from the host network namespaces of the nodes. 
+Steps must be taken to fix a CSI-with-multus issue documented
+[here](https://github.com/rook/rook/issues/8085). To summarize the issue:
+when a CephFS/RBD volume is mounted in a Pod using cephcsi and then the CSI CephFS/RBD plugin is
+restarted or terminated (e.g. by restarting or deleting its DaemonSet), all operations on the volume
+become blocked, even after restarting the CSI pods. The only workaround is to restart the node where
+the cephcsi plugin pod was restarted.
+
+When deploying a CephCluster resource configured to use multus networks, a multus-connected network
+interface will be added to the host network namespace for all nodes that will run CSI plugin pods.
+
+The Rook-Ceph Operator's CSI controller creates a DaemonSet configured to use the
+`network.selectors.public` network specified for the CephCluster. This DaemonSet runs on all the
+nodes that will have CSI pods. Its Pods exist to "hold" a particular network interface that the CSI
+pods can reliably connect to. These Pods will be referred to as the "holder" Pods.
+
+TODO: VERIFY THIS, MAYBE IT'S UNECCESSARY: These Pods should only be stopped and restarted when a
+node is stopped so that volume operations do not become blocked. The Rook-Ceph Operator's CSI
+controller should place a finalizer on the Pods so that they cannot be accidentally removed, and we
+should make note that even if the holder DaemonSet is updated, the Pods will only be changed on a
+node reboot.
+
+Furthermore, a new Rook-specific CSI sidecar will run in CSI plugin Pods alongside the Ceph CSI
+plugin process. This sidecar will watch for holder Pods, and when a holder Pod is created or
+updated, it will move the public network from the Pod's network namespace to the host node's network
+namespace. In this way, even if the CSI pods are restarted, the network interface will not change,
+and the operations to the Ceph volume will not be blocked.
+
+If a holder Pod is removed, the sidecar should remove the interface from the host's network
+namespace. Similarly, if the sidecar container is stopped, it should move all holder Pod namespaces
+from the host's network namespace back to the holder Pod's network namespace.
+
+The sidecar must:
+- be a privileged container
+- have SYS_ADMIN and NET_ADMIN capabilities
+- be on the host network namespace
+- have access to the /var/run/netns directory
+
+OLD: If an error occurs during the setup job, the job will clean up after itself, and the holder pod will
+restart. The holder pod will receive a fresh interface from multus, and try the migration again.
+NEW: If an error occurs in the sidecar during network migration, it will fail, causing the CSI
```

Comment:
Agreed, if the multus migration fails, the CSI driver won't be able to function so restarting and trying again makes sense.